bio-qiime-pipeline/src/primer_trimming.py

```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
+from zipfile import ZipFile, ZIP_DEFLATED
+
+import logging
+
+from coretex import Experiment, CustomDataset, CustomSample, folder_manager
+from coretex.bioinformatics import cutadaptTrim
+
+from .utils import loadPairedEnd
+
+
+def forwardMetadata(sample: CustomSample, outputDataset: CustomDataset) -> None:
+    sample.unzip()
+    metadataZip = folder_manager.temp / "metadata.zip"
+    with ZipFile(metadataZip, 'w', ZIP_DEFLATED) as archive:
+        for path in sample.path.iterdir():
+            archive.write(path, path.name)
+
+    if CustomSample.createCustomSample(sample.name, outputDataset.id, metadataZip) is None:
+        raise RuntimeError(">> [Microbiome analysis] Failed to forward metadata to the output dataset")
+
+
+def loadSingleEnd(sample: CustomSample) -> Path:
+    sample.unzip()
```

Comment:
add newline after unzip, please do so following the pattern everywhere I have not left a discussion